changes.go

```diff
@@ -162,27 +162,44 @@ func (g *Commands) resolveChangeListRecv(
 			return
 		}
 	}
-
-	// TODO: limit the number of active tasks for children lookups
 	dirlist := merge(remoteChildren, localChildren)
+
+	// Arbitrary value. TODO: Calibrate or calculate this value
+	chunkSize := 100
+	srcLen := len(dirlist)
+	chunkCount, remainder := srcLen/chunkSize, srcLen%chunkSize
+	i := 0
+
+	if remainder != 0 {
+		chunkCount += 1
+	}
+
 	var wg sync.WaitGroup
-	wg.Add(len(dirlist))
-	for _, l := range dirlist {
-		go func(wg *sync.WaitGroup, isPush bool, cl *[]*Change, p string, l *dirList) {
+	wg.Add(chunkCount)
```

Comment:
Seems simple enough, and yes len(dirlist)/100 so no parallelism for <100. I was just working on a worker routine count based solution, but will push it up later. I'll get a proper solution in for release 5. For now am off to bed. I'll be making a couple of changes later on when I get up. Then you can compare solutions and performance.